### PR TITLE
bluestore: set default wal size; improve weird implicit block file creation logic

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by dachary** [comment]
<pre>
Reviewed-by: Loic Dachary &lt;ldachary@redhat.com>
</pre>

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -1608,29 +1608,36 @@ int BlueStore::_setup_block_symlink_or_file(
...
+	  derr << __func__ << " failed to resize " << name << " file to "
+	       << size << ": " << cpp_strerror(r) << dendl;
+	  VOID_TEMP_FAILURE_RETRY(::close(fd));
+	  return r;
+	}
+	dout(1) << __func__ << " resized " << name << " file to "
+		<< pretty_si_t(size) << "B" << dendl;
+      }
+      VOID_TEMP_FAILURE_RETRY(::close(fd));
+    } else if (fd < 0) {
```
Comment (xiexingguo) [inline src/os/bluestore/BlueStore.cc]: Just __else__ is enough

File: src/os/bluestore/BlueStore.cc [hunk 2]
```diff
@@ -1608,29 +1608,36 @@ int BlueStore::_setup_block_symlink_or_file(
...
+      int r = -errno;
+      if (r != -ENOENT) {
+	derr << __func__ << " failed to stat " << name << " file: "
```
Comment (xiexingguo) [inline src/os/bluestore/BlueStore.cc]: failed to "open" instead of stat

Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: fixed

**Pull request merged**